{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/tests.py

```diff
@@ -71,12 +72,19 @@ def test_user_can_login(test_user):
     assert res.status_code == 200
 
 
+@pytest.mark.use_requests
 @pytest.mark.django_db
-def test_password_reset(test_user, client):
-    test_user.set_password("testing123")
-    test_user.save()
-    context = test_user.reset_password_context()
-    password_reset_url = f"/api/password/reset/confirm/{ context['user'].id }/{ context['token'] }/"
+def test_password_reset(caplog, test_user, client):
+    # fake our API call to the view that generates an email for the user to reset their password
+    rf = RequestFactory()
+    post_request = rf.post("api/password/reset/", {"email": test_user.email})
+    request_reset_link(post_request)
+
+    # Grab from the logs the actual URL link we would send to the user
+    password_reset_creds = caplog.text.split("password/reset/confirm/")[1].split('"')[0]
+    password_reset_url = f"/api/password/reset/confirm/{password_reset_creds}/"
+
+    # Verify the link works for reseting the password
     response = client.post(password_reset_url, data={"password": "new_password"}, format="json")
     assert response.status_code == 200
```

Comment:
Making this test smarter to catch more password reset bugs. When email sending isn't configured, the email contents are sent to logs for dev convenience.
This test looks at those logs to confirm that the actual URL we generate is a valid one.

Reply:
Also notice the use of the `use_requests` marker here.
When building the HTML email we inline the CSS. The library we use to do that has to call the internet to get the styles.

I think the use of this marker could be treated like a code smell. Allow it for the moment, but ideally we would come back and properly mock that call to the internet to improve speed and stability of the test suite.